gems/aws-sdk-core/spec/aws/plugins/retry_errors_spec.rb

```diff
@@ -291,6 +291,54 @@ module Plugins
           # Have to run the method each time because there are no failures
           successes.each { |success| handle_with_retry([success]) }
         end
+
+        it 'verifies success and throttling behavior' do
+          client_rate_limiter = config.client_rate_limiter
+          client_rate_limiter.instance_variable_set(:@last_throttle_time, 0)
+          # Needs to be smaller than 't' in the iterations
+          client_rate_limiter.instance_variable_set(:@last_tx_rate_bucket, 0)
+          client_rate_limiter.instance_variable_set(:@last_max_rate, 0)
+
+
+          def success(timestamp, measured_tx_rate, fill_rate)
```

Comment:
We might as well add these to the helper again and use them in the other tests

Reply:
They are actually testing different expectations (eg the cubic calculations for success test is expecting calculated rate and this "e2e" test is looking at fill rate and measured_tx_rate.  I think it makes more sense to leave these local to this test only.